ttyplot.c

```diff
@@ -203,27 +201,43 @@ static void draw_line(int x, int ph, int l1, int l2, cchar_t *c1, cchar_t *c2,
 }
 
 static void plot_values(int ph, int pw, double *v1, double *v2, double max, double min,
-                        int n, cchar_t *pc, cchar_t *hce, cchar_t *lce, double hm) {
+                        int n, cchar_t *pc, cchar_t *hce, cchar_t *lce, double hardmax,
+                        double hardmin) {
     const int first_col = 3;
     int i = (n + 1) % pw;
     int x;
-    max -= min;
-
-    for (x = first_col; x < first_col + pw; x++, i = (i + 1) % pw)
-        draw_line(x, ph,
-                  (v1[i] > hm)    ? ph
-                  : (v1[i] < min) ? 1
-                                  : (int)(((v1[i] - min) / max) * (double)ph),
-                  (v2[i] > hm)    ? ph
-                  : (v2[i] < min) ? 1
-                                  : (int)(((v2[i] - min) / max) * (double)ph),
-                  (v1[i] > hm)    ? hce
-                  : (v1[i] < min) ? lce
-                                  : pc,
-                  (v2[i] > hm)    ? hce
-                  : (v2[i] < min) ? lce
-                                  : pc,
+    int l1, l2;
+
+    for (x = first_col; x < first_col + pw; x++, i = (i + 1) % pw) {
+        /* suppress drawing uninitialized entries */
+        if (! v1 || isnan(v1[i]))
+            continue;
+
+        if (v1[i] > hardmax)
+            l1 = ph;
+        else if (v1[i] < hardmin)
+            l1 = 0;
+        else
+            l1 = lrint((v1[i] - min) / (max - min) * ph);
+
+        if (! v2 || isnan(v2[i]))
+            l2 = 0;
+        else if (v2[i] > hardmax)
+            l2 = ph;
+        else if (v2[i] < hardmin)
+            l2 = 0;
```

Comment:
```suggestion
        else if (v2[i] < hardmin)
            l2 = 1;
```
For symmetry with the case `v1[i] < hardmin`.